### PR TITLE
Allow edits to users with empty name fields

### LINKED ISSUE #40750: "Cannot read properties of undefined (reading 'id')" when deleting custom user properties
### Describe the bug

If I want to delete user properties I get that error (users were created via sso without first or last name)

### To Reproduce

1) create a user via SSO without first name or last name
2) add user properties
3) save
4) then try to delete those

### Expected behavior

admins should be able to delete user properties

### Logs

middleware.log :: PUT /api/user/2 400 1.7 ms (0 DB calls) 
{:errors {:first_name "nullable value must be a non-blank string."},
 :specific-errors {:first_name ["should be at least 1 characters, received: \"\"" "non-blank string, received: \"\""]}}

2024-03-28 17:01:12,834 DEBUG middleware.log :: GET /api/health 200 184.0 µs (0 DB calls) App DB connections: 1/13 Jetty threads: 4/50 (2 idle, 0 queued) (39 total active threads) Queries in flight: 0 (0 queued)
2024-03-28 17:01:18,981 DEBUG middleware.log :: PUT /api/user/2 400 5.4 ms (1 DB calls) 
{:errors {:first_name "Editing first name is not allowed for SSO users."}}

### Information about your Metabase installation

```JSON
v49
```

### Severity

P3

### Additional context

workaround: users should have first or last name

### LINKED ISSUE #46446: Not possible to update user details without both the first and the last name being set
### Describe the bug

We've removed the requirement for the user's first and last name a few years ago.
So it's totally ok to create a user with only an email.

Trying to update such user using the UI will fail because the frontend sends empty strings instead of `null` for first/last name values.

### To Reproduce

1. Create new user with only an email
2. Click on `...` and edit user
3. Try to set only the first name, only the last name or to change anything else about this user (what matters is that we trigger `PUT` request)
4. The request will fail `400` because frontend sends empty strings

The response if we try to update only the last name:
```
{
  "first_name": "nullable value must be a non-blank string."
}
```

### Expected behavior

Frontend should send `null` as a value when first and/last names are missing.

### Logs

_No response_

### Information about your Metabase installation

```JSON
local dev, `master`, da92a62, H2, Sample Database
```

### Severity

P1

### Additional context

It is confirmed that this is a pure FE bug. The BE expects either `null` or non-empty string.
image

This was still possible in v47 (I tried 47.6), but it broke in v48.

### PR DESCRIPTION
**Pull request by iethree**
Closes https://github.com/metabase/metabase/issues/46446
Closes https://github.com/metabase/metabase/issues/40750

### Description

Our API allows metabase users to have an empty firstname and lastname. However, our user edit form was sending empty strings instead of nulls, which our API does not allow. This updates the admin user edit form to allow two previously unhandled cases:
1. Editing user details when they are missing either first name, last name or both (reported in the issue)
2. Removing a user's first name, last name, or both

Screen Shot 2024-08-05 at 5 40 23 PM

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
Can you please check this potentially related issue at the same time?
https://github.com/metabase/metabase/issues/40750

**Comment by iethree** [comment]
> Can you please check this potentially related issue at the same time? #40750

@nemanjaglumac thanks for calling this out! I accidentally fixed this too! I'll add a test that covers attributes as well 👍

File: frontend/src/metabase/admin/people/containers/EditUserModal.unit.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      login_attributes: {},
+    });
+  });
+
+  describe("users with empty name fields (metabase#46446)", () => {
+    it("can update the first name of a user with no last name", async () => {
+      setup({
+        userData: createMockUserInfo({
+          id: 11,
+          first_name: "Prince",
```
Comment (rafpaf) [inline frontend/src/metabase/admin/people/containers/EditUserModal.unit.spec.tsx:81]: Added via Giphy

File: frontend/src/metabase/admin/people/containers/EditUserModal.unit.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+          last_name: null,
+          email: "name@example.com",
+        }),
+      });
+
+      const firstNameField = await screen.findByLabelText("First name");
+      const submitButton = await screen.findByText("Update");
+
+      await userEvent.clear(firstNameField);
+      await userEvent.type(firstNameField, "Madonna");
```
Comment (rafpaf) [inline frontend/src/metabase/admin/people/containers/EditUserModal.unit.spec.tsx:91]: Added via Giphy

File: frontend/src/metabase/admin/people/containers/EditUserModal.unit.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        login_attributes: {},
+      });
+    });
+
+    it("can remove a user's name", async () => {
+      setup({
+        userData: createMockUserInfo({
+          id: 11,
+          first_name: "Simon",
+          last_name: "Garfunkel",
```
Comment (rafpaf) [inline frontend/src/metabase/admin/people/containers/EditUserModal.unit.spec.tsx:140]: brutal

**Review by rafpaf — APPROVED** [review]

**Pull request merged**